api.planx.uk/webhooks/sanitiseApplicationData/operations.ts

```diff
@@ -0,0 +1,214 @@
+import { gql } from "graphql-request";
+import { subMonths } from "date-fns";
+
+import { Operation, OperationResult } from "./types";
+import { adminGraphQLClient } from "../../hasura";
+import { runSQL } from "../../hasura/schema";
+
+const RETENTION_PERIOD_MONTHS = 6;
+export const getRetentionPeriod = () => subMonths(new Date(), RETENTION_PERIOD_MONTHS);
+
+/**
+ * List of data sanitation operations
+ * XXX: Analytics logs do not contain application data
+ */
+ export const getOperations = (): Operation[] => ([
+  // Raw application data
+  sanitiseLowcalSessions,
+
+  // Audit records
+  sanitiseSessionBackups,
+  sanitiseUniformApplications,
+  sanitiseBOPSApplications,
+  sanitiseReconciliationRequests,
+
+  // Event logs
+  deleteHasuraEventLogs,
+]);
+
+export const operationHandler = async (operation: Operation): Promise<OperationResult> => {
+  let operationResult: OperationResult = {
+    operationName: operation.name,
+    status: "processing"
+  };
+
+  try {
+    const result = await operation()
+    operationResult = {
+      ...operationResult,
+      status: "success",
+      count: result.length,
+    };
+  } catch (error) {
+    operationResult = {
+      ...operationResult,
+      status: "failure",
+      errorMessage: (error as Error).message,
+    }
+  };
+
+  return operationResult;
+}
+
+export const sanitiseLowcalSessions: Operation = async () => {
+  const mutation = gql`
+    mutation SanitiseLowcalSessions($retentionPeriod: timestamptz) {
```

Comment:
I'm also interested in this question! Related to prior comment about sanitising after a week and hard-deleting after 6 months. 

I think we could communicate this policy as:
- We have anonymous analytics for a service over _all time_
- We have fine-grained save&return/reconciliation/submission analytics _per session_ for the _last six months_
  - Eg anonymous analytics could tell you an app was submitted/reached Confirmation page, but fine-grained audit tables could tell you _where_ it was submitted to. I think the latter is perhaps still more a function of troubleshooting anyways rather than full service stats since Send destinations don't change frequently. 

Probably one for Alastair & partners to help confirm!